### DISCUSSION TITLE
Export Module-Type

### DISCUSSION THREAD

**Discussion by ciroiriarte** [netbox-community/devicetype-library#1343] [Q&A] [answered]
Hello!, anybody can confirm if there's a "module-type export to yaml" feature somewhere?. In order to model yaml files to collaborate, I would like to confirm how should some things be modeled in a yaml file.

**Comment by danner26** [comment]
Hey @ciroiriarte there is not currently a way to export these via the UI, but I am looking into if it is possible to do so via [pynetbox](https://github.com/netbox-community/pynetbox) or similar

**Reply by danner26** [reply, marked as answer]
Alright so I have done some digging. Using pynetbox you can pull the module type/device type like so:

1. You can acquire the specific module like so: `netbox.dcim.module_types.filter(model='<MODULE_MODEL>)` where `MODULE_MODEL` is the slug of the module/device. Let save that to a variable such as `module`
2. Then, we can store the RecordSet as a list like so: `mod_var = list(module)` where `mod_var` is an arbitrary variable name
3. Lastly, we can then print out the module representiation (or store them in a var) like so: `mod_var[0].__dict__`

This will give you a JSON dataset. You can then either use this or convert it to YAML and it should be importable. 

**\*\*WARNING\*\***: The import part is not tested. You are welcome to comment back if something doesnt work
